src/ui/HeaderRenderer.ts

```diff
@@ -336,6 +349,45 @@ export default class HeaderRenderer {
       provider.takeSnapshot(d);
       (<MouseEvent>d3.event).stopPropagation();
     });
+
+    //Heatmap Sort
+    $node.filter((d) => d instanceof HeatmapColumn).append('i').attr('class', 'fa fa-sort').attr('title', 'Sort By').on('click', function (d) {
+      sortDialogHeatmap(<HeatmapColumn>d, d3.select(this.parentNode.parentNode));
+      (<MouseEvent>d3.event).stopPropagation();
+    });
+
+    $node.filter((d) => d instanceof SparklineColumn).append('i').attr('class', 'fa fa-sort').attr('title', 'Sort By').on('click', function (d) {
+      sortDialogSparkline(<SparklineColumn>d, d3.select(this.parentNode.parentNode));
+      (<MouseEvent>d3.event).stopPropagation();
+    });
+
+
+    $node.filter((d) => d instanceof BoxplotColumn).append('i').attr('class', 'fa fa-sort').attr('title', 'Sort By').on('click', function (d) {
+      sortDialogBoxplot(<BoxplotColumn>d, d3.select(this.parentNode.parentNode));
+      (<MouseEvent>d3.event).stopPropagation();
+    });
+
+
+    $node.filter((d) => d instanceof ThresholdColumn).append('i').attr('class', 'fa fa-sort').attr('title', 'Sort By').on('click', function (d) {
+      sortDialogThresholdBar(<ThresholdColumn>d, d3.select(this.parentNode.parentNode));
+      (<MouseEvent>d3.event).stopPropagation();
+    });
+
+
+    $node.filter((d) => d instanceof VerticalBarColumn).append('i').attr('class', 'fa fa-sort').attr('title', 'Sort By').on('click', function (d) {
+
+      sortDialogVerticalBar(<VerticalBarColumn>d, d3.select(this.parentNode.parentNode));
+      (<MouseEvent>d3.event).stopPropagation();
+    });
+
+
+    //Renderer Change
+    $node.filter((d) => d instanceof Column).append('i').attr('class', 'fa fa-exchange').attr('title', 'Change Visualization').on('click', function (d) {
```

Comment:
all elements in the header are columns by default + does every column really has multiple renderers to choose from?